test/benchmark_setup_teardown_cb_types_test.cc

```diff
@@ -0,0 +1,83 @@
+#include <iostream>
+#include <memory>
+
+#include "benchmark/benchmark.h"
+
+using benchmark::callback_function;
+struct Functor {
+  Functor() {}
+  Functor(const Functor& /*unused*/) {}
+  Functor(Functor&& /*unused*/) {}
+  void operator()(const benchmark::State& /*unused*/) {}
+};
+
+void BM_DoSomething(benchmark::State& state) {
+  for (auto _ : state) {
+  }
+}
+
+void function(const benchmark::State& /*unused*/) {}
+
+int main(int argc, char** argv) {
```

Comment:
this test doesn't appear to be, well, testing anything.  can you please take a look at how the other tests are structured and see about making this more useful?